blah.cpp

```diff
@@ -0,0 +1,82 @@
+#include <stdio.h>
+#include <stdbool.h>
+#include <iostream>
+#include <cstring>
+#include <cstdlib>
+#include <cctype.h>
+
+bool characteristic(char numString[], int& c);
+bool mantissa(char numString[], int& numerator, int& denominator);
+
+int main()
+{
+    char number[] = "123.456";
+    int c, n, d;
+
+    //if the conversion from C string to integers can take place
+    if(characteristic(number, c) && mantissa(number, n, d))

```

Comment:
You should consider writing separate functions instead of having all of the functionality occur directly within the main function. The main should act more as a driver which calls other functions, instead of handling all of the processing by itself.
For example: 
Main takes the input from the user, but then sends it to another function that "sanitizes" the data by removing trailing white spaces, verifies that the data is indeed valid, and perhaps strips the "-" / "+" sign from the input for later use. The main would then send this "clean" data to the other functions which will then perform the actions that your current mantissa and characteristic functions do.